shell/platform/fuchsia/flutter/vulkan_surface.cc

```diff
@@ -50,33 +60,87 @@ bool VulkanSurface::CreateVulkanImage(vulkan::VulkanProvider& vulkan_provider,
       .collection = collection_,
       .index = 0,
   };
+  
+  const VkSysmemColorSpaceFUCHSIA kDefaultColorSpace = {
+    .sType = VK_STRUCTURE_TYPE_SYSMEM_COLOR_SPACE_FUCHSIA,
+    .pNext = nullptr,
+    .colorSpace = static_cast<uint32_t>(fuchsia::sysmem::ColorSpaceType::SRGB),
+  };
+
+  const VkImageFormatConstraintsInfoFUCHSIA kDefaultFormatInfo = {
+    .sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_CONSTRAINTS_INFO_FUCHSIA,
+    .pNext = nullptr,
+    .requiredFormatFeatures = 0u,
+    .flags = 0u,
+    .sysmemFormat = static_cast<uint64_t>(fuchsia::sysmem::ColorSpaceType::INVALID),
+    .colorSpaceCount = 1,
+    .pColorSpaces = &kDefaultColorSpace,
+  };
 
-  out_vulkan_image->vk_image_create_info = {
-      .sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO,
-      .pNext = &out_vulkan_image->vk_collection_image_create_info,
-      .flags = kVulkanImageCreateFlags,
-      .imageType = VK_IMAGE_TYPE_2D,
-      .format = kVulkanFormat,
-      .extent = VkExtent3D{static_cast<uint32_t>(size.width()),
-                           static_cast<uint32_t>(size.height()), 1},
-      .mipLevels = 1,
-      .arrayLayers = 1,
-      .samples = VK_SAMPLE_COUNT_1_BIT,
-      .tiling = VK_IMAGE_TILING_OPTIMAL,
-      .usage = kVkImageUsage,
-      .sharingMode = VK_SHARING_MODE_EXCLUSIVE,
-      .queueFamilyIndexCount = 0,
-      .pQueueFamilyIndices = nullptr,
-      .initialLayout = VK_IMAGE_LAYOUT_UNDEFINED,
+  const VkImageCreateInfo kDefaultImageCreateInfo = {
+    .sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO,
+    .pNext = &out_vulkan_image->vk_collection_image_create_info,
+    .flags = 0u,
+    .imageType = VK_IMAGE_TYPE_2D,
+    .format = VK_FORMAT_UNDEFINED,
+    .extent = VkExtent3D{static_cast<uint32_t>(size.width()),
+                          static_cast<uint32_t>(size.height()), 1},
+    .mipLevels = 1,
+    .arrayLayers = 1,
+    .samples = VK_SAMPLE_COUNT_1_BIT,
+    .tiling = VK_IMAGE_TILING_OPTIMAL,
+    .usage = kVkImageUsage,
+    .sharingMode = VK_SHARING_MODE_EXCLUSIVE,
+    .queueFamilyIndexCount = 0,
+    .pQueueFamilyIndices = nullptr,
+    .initialLayout = VK_IMAGE_LAYOUT_UNDEFINED,
+  };
+
+  constexpr size_t kNumFormats = 2u;
+  std::vector<VkImageFormatConstraintsInfoFUCHSIA> format_constraints(kNumFormats, kDefaultFormatInfo);
+  std::vector<VkImageCreateInfo> create_infos(kNumFormats, kDefaultImageCreateInfo);
+  // Immutable format is technically limited to R8G8B8A8_SRGB but
+  // R8G8B8A8_UNORM works with existing ARM drivers so we allow that
+  // until we have a more reliable API for creating external Vulkan
+  // images using sysmem. TODO(fxb/52835)
```

Comment:
this commend doesn't make sense when using sysmem. why are you adding this back?